testing/e2e/suite/comand_line_integration.go

```diff
@@ -0,0 +1,215 @@
+)
+
+var (
+	dfOpts = []cmp.Option{
+		cmp.Transformer("T1", func(in string) string {
+			s := strings.TrimSpace(in)
+			s = strings.Replace(in, "  ", " ", -1)
+			return regexp.MustCompile(`( (\d+[smhd]){1,4})[ \n$]`).ReplaceAllString(s, "0s")
+		}),
+		// ^([0-9]{1,10}[dhms]{1}){1,4}$
+	}
+	objOpts = []cmp.Option{
+		cmpopts.IgnoreFields(cartov1alpha1.Workload{},
+			"Status",
+			"GenerateName",
+			"SelfLink",
+			"UID",
+			"ResourceVersion",
+			"Generation",
+			"CreationTimestamp",
+			"DeletionTimestamp",
+			"DeletionGracePeriodSeconds",
+			"OwnerReferences",
+			"Finalizers",
+			"ZZZ_DeprecatedClusterName",
+			"ManagedFields",
+		),
+	}
+)
+
+const TestingNamespace = "apps-integration-testing"
+
+type CommandLineIntegrationTestSuite []CommandLineIntegrationTestCase
+
+type CommandLineIntegrationTestCase struct {
+	Name                      string
+	Skip                      bool
+	Focus                     bool
+	WorkloadName              string
+	Command                   CommandLine
+	ExpectedCommandLineError  bool
+	ExpectedCommandLineOutput string
+	SkipClusterValidation     bool
+	ExpectedRemoteObject      client.Object
+	IsList                    bool
+	ExpectedRemoteList        client.ObjectList
+	SleepMillisBefore         int
+	SleepMillisAfter          int
+	CheckCommandOutput        bool
+}
+
+func (ts CommandLineIntegrationTestSuite) Run(t *testing.T) {
+
+	cleanUp(t, false)
+	preapreEnvironment(t)
+	defer cleanUp(t, t.Failed())
+
+	testToRun := ts
+	focused := CommandLineIntegrationTestSuite{}
+	for _, tc := range ts {
+		if tc.Focus && !tc.Skip {
+			focused = append(focused, tc)
+		}
+	}
+	isFocused := len(focused) != 0
+	if isFocused {
+		testToRun = focused
+	}
+
+	conf, err := NewClientConfig()
+	if err != nil {
+		t.Errorf("Error creating validation config: %v", err)
+		t.FailNow()
+	}
+
+	for _, tc := range testToRun {
+		tc.Run(t, conf)
+		if t.Failed() {
+			break
+		}
+	}
+	if isFocused {
+		t.Errorf("test run focused on %d record(s), skipped %d record(s)", len(focused), len(ts)-len(focused))
+	}
+
+}
+
+func (cl CommandLineIntegrationTestCase) Run(t *testing.T, conf *ClientConfig) {
+	t.Run(cl.Name, func(t *testing.T) {
+
+		if cl.Skip {
+			t.SkipNow()
+		}
+		time.Sleep(time.Duration(cl.SleepMillisBefore) * time.Millisecond)
+		err := cl.Command.Exec()
+		// t.Logf("Command output:\n%s\n\n%v\n", cl.Command.String(), cl.Command.GetOutput())
+		if err != nil && !cl.ExpectedCommandLineError {
+			t.Errorf("unexpected error: %v: %v, ", err, cl.Command.GetOutput())
+			t.FailNow()
+		}
+
+		if err == nil && cl.ExpectedCommandLineError {
+			t.Errorf("Expected error, got nil")
+			t.FailNow()
+		}
+
+		if cl.CheckCommandOutput {
+			if diff := cmp.Diff(cl.ExpectedCommandLineOutput, cl.Command.GetOutput(), dfOpts...); diff != "" {
+				t.Errorf("%s\n(-expected, +actual)\n%s", cl.Name, diff)
+				t.FailNow()
+			}
+		}
+
+		if !cl.SkipClusterValidation {
```

Comment:
`SkipClusterValidation` are there any cases we want to skip this for some reason? I thought  cluster validation would be part of setup for e2e tests like check CRDs exist and test ns exist